file_upload_handler/src/server.js

```diff
@@ -0,0 +1,59 @@
+import express from "express";
+import multer from "multer";
+
+const PORT = process.env.PORT || 8000
+const app = express();
+
+const storage = multer.diskStorage({
+  destination: function (req, file, cb) {
+    cb(null, "./uploads");
+  },
+  filename: function (req, file, cb) {
+    const uniquepreffix = Date.now() + "-" + Math.round(Math.random() * 1e9);
+    cb(null, uniquepreffix + "-" + file.originalname);
+  },
+});
+
+function fileFilter(req, file, cb) {
+  // The function should call `cb` with a boolean
+  // to indicate if the file should be accepted
+  const acceptFileType = ["video/mp4", "image/png", "image/jpeg", "image/gif"];
+  if (acceptFileType.includes(file.mimetype)) {
+    // To accept the file pass `true`, like so:
+    cb(null, true);
+  } else {
+    cb(new Error("invalid File type\nonly accept" + acceptFileType));
+  }
+
+  //cb(null, false)
+
+  // You can always pass an error if something goes wrong:
+  //cb(new Error('I don\'t have a clue!'))
+}
+
+var upload = multer({
```

Comment:
`var` isn't usually used anymore in Javascript. Instead you should use either `const` or `let`. Rule of thumb is to use `const` by default. `let` is necessary only if you need to re-assign the variable later
e.g
```
const i = 1;
i = 2; // This will not work (causes an error), can't re-assign `i` since it was defined using const

let x = 1
x = 2 // re-assignment, this is allowed since we used let
```